db/src/Pos/DB/Block/Epoch.hs

```diff
@@ -1,30 +1,49 @@
 {-# LANGUAGE DeriveGeneric   #-}
 {-# LANGUAGE TemplateHaskell #-}
 
+-- This module contains the code for consolidating blund (block/undo) files
+-- into an epoch/index file pair.
+--
+-- There are up to 21600 blocks per epoch and these are consolidated into an
+-- epoch file and an index for lookup the blund by 'SlotId'. Once an entire
+-- epoch has been consolidated into an epoch/index file pair, the old blund
+-- files are deleted.
+--
+-- Epoch files (which have a header containing version information) consist
+-- of the concatenation of the data for all the slots within the epoch, where
+-- the data for a slot is stored as as the concatenation of:
+--   * the four characters "blnd" to mark the start of a slot (for debugging)
+--   * a 32 bit value for the length of the block
+--   * a 32 bit value for the length of the undo
+--   * the block itself
+--   * the undo itself
+--
+-- Epoch index files contain a header and then an indexed vector of the offset
+-- into the file where the data for a slot will be found. If an offset in the
+-- file has all its bits set, then that mean that there is no block/undo for
+-- that slot index.
+
```

Comment:
Suggest you add comment  here to point out that the epoch consolidation process leaves (at least) the last two epoch 'unblunded'. I had to remember that to make sense of some of the logic a bit further down (like the issue during initial sync)